src/com/google/enterprise/adaptor/database/ResponseGenerator.java

```diff
@@ -386,20 +404,126 @@ public void generateResponse(ResultSet rs, Response resp)
+              Writer writer = new OutputStreamWriter(out, UTF_8)) {
+            if (reader != null) {
+              copy(reader, writer);
+            }
+          }
+          break;
+        case Types.VARBINARY:
+          try (ByteArrayInputStream in =
+              new ByteArrayInputStream(rs.getBytes(index))) {
+            IOHelper.copyStream(in, out);
+          }
+          break;
+        case Types.LONGVARBINARY:
+          try (InputStream in = rs.getBinaryStream(index)) {
+            if (in != null) {
+              IOHelper.copyStream(in, out);
+            }
+          }
+          break;
+        case Types.LONGNVARCHAR:
+          try (Reader reader = rs.getNCharacterStream(index);
+              Writer writer = new OutputStreamWriter(out, UTF_8)) {
+            if (reader != null) {
+              copy(reader, writer);
+            }
+          }
+          break;
+        case Types.CLOB:
+          Clob clob = rs.getClob(index);
+          if (clob != null) {
+            try (Reader reader = clob.getCharacterStream();
+                Writer writer = new OutputStreamWriter(out, UTF_8)) {
+              copy(reader, writer);
+            } finally {
+              try {
+                clob.free();
+              } catch (java.sql.SQLFeatureNotSupportedException
+                  | UnsupportedOperationException unsupported) {
+                // let JVM garbage collection deal with it
+                log.log(Level.FINEST, "Error closing clob", unsupported);
+              }
+            }
+          }
+          break;
+        case Types.NCLOB:
+          NClob nclob = rs.getNClob(index);
+          if (nclob != null) {
+            try (Reader reader = nclob.getCharacterStream();
+                Writer writer = new OutputStreamWriter(out, UTF_8)) {
+              if (reader != null) {
+                copy(reader, writer);
+              }
+            } finally {
+              try {
+                nclob.free();
+              } catch (java.sql.SQLFeatureNotSupportedException
+                  | UnsupportedOperationException unsupported) {
+                // let JVM garbage collection deal with it
+                log.log(Level.FINEST, "Error closing nclob", unsupported);
+              }
+            }
+          }
+          break;
+        case Types.BLOB:
+          Blob blob = rs.getBlob(index);
+          if (blob != null) {
+            try (InputStream in = blob.getBinaryStream()) {
+              IOHelper.copyStream(in, out);
+            } finally {
+              try {
+                blob.free();
+              } catch (java.sql.SQLFeatureNotSupportedException
+                  | UnsupportedOperationException unsupported) {
+                // let JVM garbage collection deal with it
+                log.log(Level.FINEST, "Error closing blob", unsupported);
+              }
+            }
+          }
+          break;
+        case Types.SMALLINT:
```

Comment:
List in a logical order, and BIGINT and REAL are missing: integers by size, floats by size, chars (as-is).